rfc/rfc-48/rfc-48.md

```diff
@@ -0,0 +1,174 @@
+
+Plan strategy class fetches all the file groups from the eligible partition paths and filters out file groups that are not eligible for LogCompaction. File groups are excluded from LogCompaction if any of the following properties holds true.
+1. File groups are already part of a major compaction plan or clustering operation.
+2. File groups whose file slices or log files are part of a LogCompaction operation.
+
+####Note:
+Here Logcompaction can also be scheduled on a pending compaction operation. Since, pending compaction operation 
+creates a new file slice and delta commits can still come into the latest file slice without compaction being done.
+
+
+#### Disadvantages:
+
+LogCompaction on a file group can create large log blocks that are self-sufficient enough and do not require additional stitching operation to be performed, i.e. as shown in the figure below larger log blocks can be created by LogCompaction. In this scenario it might be better to stitch log blocks(5,6) and ignore the ones that are already large enough to reduce write amplification caused due to rewriting huge log blocks again and again.
+
+To overcome the above disadvantage instead of blindly merging all the log blocks for a file group, minor compaction can be scheduled only for a subset of consecutive log blocks that have block sizes smaller and exceed a threshold count. That way only the required log blocks are merged together.
+
+
+### LogCompaction Execution:
+Information regarding the list of log blocks to be merged is part of the LogCompaction plan. 
+Execution strategy reads through the plan and merges the log blocks together and creates a new block.
+
+### Cleaner:
+Merged LogBlocks are only cleaned once a complete or major compaction is executed on the file group to form a new base file. So, there won’t be any changes required from the cleaner service. Instead of log blocks if there are log files, then cleaner can clean the replaced log files.
+
+### Archival:
+Similar to compaction, archival can be performed for LogCompaction commits as well.
+
+## Rollout/Adoption Plan
+
+- What impact (if any) will there be on existing users?
+  - LogCompaction action is not enabled by default.
+  - Users have to explicitly add the configs to use this feature so existing users are not impacted.
+  - Users can enable this feature to reduce write amplification caused due to frequent compaction on the MOR table.
+- If we are changing behavior how will we phase out the older behavior?
+    - We are not changing the older behavior, compaction is still required to create newer versions of the base files.
+  - LogCompaction is used to reduce the no. of log blocks to consider while reading.
+  - By using this feature, frequency of the compaction operation can be reduced.
+- If we need special migration tools, describe them here.
```

Comment:
don't the readers need to enable scanV2 before writers starting to write? so, we need to migrate readers first.. and then migrate writers if I am not wrong.